src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.PullMember)), Shared]
+    internal class CSharpPullMemberUpCodeRefactoringProvider : AbstractPullMemberUpRefactoringProvider
+    {
+        protected override bool IsSelectionValid(TextSpan span, SyntaxNode userSelectedSyntax)
+        {
+            var identifier = GetIdentifier(userSelectedSyntax);
+            if (identifier == default)
+            {
+                return false;
+            }
+            else
+            {
+                return (identifier.FullSpan.Contains(span) && span.Contains(identifier.Span)) ||  
+                    (identifier.Span.Contains(span) && span.Length == 0);
+            }
+        }
+
+        private SyntaxToken GetIdentifier(SyntaxNode userSelectedSyntax)
+        {
+            switch (userSelectedSyntax)
+            {
+                case MemberDeclarationSyntax memberDeclarationSyntax:
```

Comment:
note: TypeDeclarations are also MemberDeclarations.  Make a test that validates that we don't do anythin gwrong when the user is on the name of a type nested in another type.

Reply:
was this done?

Reply:
was this done?

Reply:
So I change the sequence of the check. Before we check the syntax here, it will first check the symbol's kind to make sure it is method, field, event or property. So the input won't contains any nested type. I can add one unit test to make sure it won't show anything and put some comments here.

Reply:
A unit test would be good.  Specifically, because it means that if we change this logic in the future, but miss that realization, we won't end up with bad behavior unintentionally.  Thanks!